compiler/rustc_mir_transform/src/errors.rs

```diff
@@ -141,10 +167,25 @@ impl<'a> DecorateLint<'a, ()> for UnsafeOpInUnsafeFn {
         let desc = diag
             .handler()
             .expect("lint should not yet be emitted")
-            .eagerly_translate_to_string(self.details.label(), [].into_iter());
+            .eagerly_translate_to_string(self.details.clone().label(), [].into_iter());
         diag.set_arg("details", desc);
-        diag.span_label(self.details.span, self.details.label());
-        diag.note(self.details.note());
+        diag.span_label(self.details.clone().span, self.details.clone().label());
```

Comment:
I think you don't need to clone details to obtain the span, just to obtain the label.

Reply:
It's necessary, if I don't clone it then will be an error here